code/trace.c

```diff
@@ -1561,7 +1530,15 @@ Res TraceScanFormat(ScanState ss, Addr base, Addr limit)
    * ss->formatScan. */
   ss->scannedSize += AddrOffset(base, limit);
 
-  return ss->formatScan(&ss->ss_s, base, limit, ss->formatScanClosure);
+  switch (ss->callbackType)
+  {
+      default:
+          NOTREACHED;
```

Comment:
Add a `return ResFAIL` here so that we can in theory continue from the assertion.